src/blog/2024/02/postgresql-with-node-red.md

```diff
@@ -0,0 +1,231 @@
+```sql
+-- Retrieve all data from the product_data table
+SELECT * FROM product_data;
+```
+!["Retriving all product data from database"](./images/postgresql_with_node-red_retrive_data.png "Retriving all product data from database")
+
+# Updating product data to the Database
+!["Adding form to update product data"](./images/postgresql_with_node-red_form2.png "adding form to update product data")
+
+In this section, we will add a form interface to collect the product ID and the new stock value for the update process. Feel free to select other data fields that you need to update. To achieve this, we will add a form interface using Dashboard 2.0. Additionally, we will interact with the database using the same PostgreSQL node that we have used so far in this guide.
+
+1. Drag a Ui-Form widget onto the canvas and create a new UI group for it.
+2. Add elements for product id and stock in the form widget and give it a name, label, and select type.
+3. Drag a function node onto Canvas and paste the following script.
+
+```javascript 
+// Destructure the properties from msg.payload
+const { id, stock } = msg.payload;
+// Create an array containing id and stock
+// The order of the array items in msg.params will correspond to the placeholders in the SQL query
+// For example, $1 will be replaced by the value of id, $2 will be replaced by the value of stock
+msg.params = [id, stock];
+return msg;
+```
+4. Drag a PostgreSQL node on canvas, click on that node and paste the following SQL command into the query input field.
+
+```sql 
+-- UPDATE statement to modify data in the product_data table
+
+UPDATE product_data 
+-- Specifies the table to be updated (product_data)
+
+SET 
+    stock =  $2
+-- Sets the value of the "stock" column to the value represented by the parameter $2.
+-- The value to be set is typically provided externally, In our context we getting this parameter by "msg.params"
+
+WHERE id = $1;
+-- Specifies the condition that must be met for the update to occur.
+-- In this case, it updates rows where the "id" column matches the value represented by the parameter $1.
+```
+
+!["Updating product data to the database"](./images/postgresql_with_node-red_update_data.png "Updating product data to the database")
+
+5. Connect Ui-form’s output to function node’s input, and function node's output to the PostgreSQL node’s input.
+
+# Deleting product data from the database
+
+!["Deleting product data to the database"](./images/postgresql_with_node-red_form3.png "Deleting product data to the database")
+
+In this section, we'll cover how to delete product data from the database. we will use Dashboard 2.0's form interface to collects essential information like the product id and name. While the product id alone is sufficient to delete a product from the database, we include the product name as an additional precaution to prevent accidental deletion of product data.
```

Comment:
```suggestion
In this section, we'll cover how to delete product data from the database. We will use Dashboard 2.0's form interface to collect essential information like the product id and name. While the product id alone is sufficient to delete a product from the database, we include the product name as an additional precaution to prevent accidental deletion of product data.
```